solr/core/src/java/org/apache/solr/cloud/ZkConfigSetService.java

```diff
@@ -151,6 +151,19 @@ public boolean checkConfigExists(String configName) throws IOException {
     }
   }
 
+  @Override
+  public boolean checkConfigExistsWithSolrConfigXml(String configName) throws IOException {
+    try {
+      Boolean existsConfig = zkClient.exists(CONFIGS_ZKNODE + "/" + configName, true);
+      if (existsConfig == null || existsConfig.booleanValue() == false) return false;
+      Boolean existsSolrConfigXml = zkClient.exists(CONFIGS_ZKNODE + "/" + configName + "/solrconfig.xml", true);
+      return existsSolrConfigXml;
```

Comment:
If there is a `existsConfig == null` check above, there should be one here as well, otherwise there will be a NPE.

Reply:
Good point, fixed.

Reply:
I'm missing something; why two calls to ZK just to check that a config exists?  If solrconfig.xml is now necessary, then why check the parent?

Reply:
Fxied